src/keep-current-value.js

```diff
@@ -48,7 +60,7 @@ function keepCurrentValue() {
 
   function link(scope, element, attrs, controller) {
     var viewValue = getViewValueFromElement(element);
-    if ( viewValue ) {
+    if ( viewValue !== undefined ) {
```

Comment:
Since the value `false` can be returned if the checkbox isn't checked and no ng-false-value is set, `viewValue` needs a more precise comparsion.